src/python/pants/backend/codegen/thrift/python/py_thrift_namespace_clash_check.py

```diff
@@ -33,6 +34,9 @@ def register_options(cls, register):
     # TODO: deprecate the --strict option in a future release!
     register('--strict', type=bool, default=False, fingerprint=True,
              help='Whether to fail the build if thrift sources have invalid python namespaces.')
+    register('--dump-invalid-thrift-source-errors-inline', type=bool, default=True,
```

Comment:
Should `strict` be converted into a ternary instead? `fail, warn, ignore`? In reality, if `strict=False` and `dump-invalid-thrift-source-errors-inline=False`, it's a tree falling in a forest.

Reply:
Well, the reason the `TODO` is there is because I think we do want to eventually require that all sources of `python_thrift_library()`s will declare a `namespace py`, so I would like to avoid the allowing the `ignore` and just allow also abbreviating the error output so that it doesn't annoy users before all the targets in the repo are fixed.

Reply:
I don't think that that changes my suggestion. But I won't block on it.

Reply:
The direct response would be: if we want to eventually make this always error, having an `ignore` value doesn't seem to make sense to me. It's not clear to me what the tree falling in a forest refers to here.

Reply:
I would feel a bit more comfortable with this discussion if you were willing to do a followup PR that marks `--strict` as deprecated so that we have less time where it's possible to have `--strict=False` and `dump-invalid-thrift-source-errors-inline=False` as well.

Won't block on this at all though.

Reply:
It is extremely unlikely that anyone will see the warning or go looking for the file that is created. Thus, there is a question of whether it will "make a sound" (the tree falling in the forest case).